### PR TITLE
Simulation: Execution trace (PC/Stack/Scratch) support

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="master"
+SDK_TESTING_BRANCH="simulate-exec-trace-stack-scratch"
```
Comment (ahangsu) [inline .test-env:3]:
```suggestion
SDK_TESTING_BRANCH="master"
```

**Review by jasonpaulos — APPROVED** [review]
Looks good besides the one comment

File: tests/cucumber/steps/steps.js [hunk 2]
```diff
@@ -4815,6 +4815,126 @@ module.exports = function getSteps(options) {
     }
   );
 
+  Then(
+    'I allow exec trace options {string} on that simulate request.',
+    async function (execTraceOptions) {
+      const optionList = execTraceOptions.split(',');
+
+      assert.ok(this.simulateRequest);
+      // TODO: check if this is the best practice
```
[the remark this answers is not in the document]
Comment (ahangsu) [reply tests/cucumber/steps/steps.js]: I was checking if `this.simulateRequest` is not undefined or null, seems to be right here?

[the remark this answers is not in the document]
Comment (jasonpaulos) [reply tests/cucumber/steps/steps.js]: Yeah that looks right

**Review by bbroder-algo — APPROVED** [review]

**Comment by bbroder-algo** [comment]
A bunch of cucumber errors around "currency-greater-than"

**Comment by bbroder-algo** [comment]
I think #807 is blocking

**Pull request merged**